protocol/client_test.go

```diff
@@ -0,0 +1,174 @@
+package protocol
+
+import "testing"
+
+var (
+	uname = "alice"
+	key   = []byte("key")
+)
+
+func TestVerifyRegistrationResponseWithTB(t *testing.T) {
+	d, pk := NewTestDirectory(t, true)
+
+	cs := NewConiksClient(d.LatestSTR().Signature, true, pk)
+
+	res, _ := d.Register(&RegistrationRequest{
+		Username: uname,
+		Key:      key})
+	if cs.Verify(res, uname, key); cs.VerificationResult != Passed && cs.ProofType != ProofOfAbsence {
```

Comment:
As I mentioned above, I think this check should be part of `cs.Verify()`. And similarly for the other proof type checks in client_test.go.